CppExport/src/dependency_analysis/DependencyAnalyzer.cpp

```diff
@@ -89,4 +89,43 @@ QList<DependencyComposite*> DependencyAnalyzer::mergeUnits(QList<DependencyUnit*
 	return nameToCompositeMap.values();
 }
 
+template <class T>
+QList<T*> DependencyAnalyzer::topologicalSort(QHash<T*, QSet<T*>> graph)
+{
+	QHash<T*, QSet<T*>> reverseGraph;
+	for (auto it = graph.begin(); it != graph.end(); it++)
+		for (auto dependency : it.value())
+			reverseGraph[dependency].insert(it.key());
+
+	QList<T*> toBeProcessed;
```

Comment:
hm I wanted to get the dependency names out of this routine since it's purpose is independent of the concept of dependencies. I forgot to get rid of some dependency related names though.